### ISSUE TITLE
Making linting errors break build

### ISSUE THREAD

**Issue by boedlen** [JoshuaKGoldberg/TSLint.MSBuild#29]
Hi Joshua,

I'm trying to make linting errors break my build, and i've managed to do it by changing:

`<Exec 
      Command="&quot;$(TSLintNodeExe)&quot; --harmony &quot;$(TSLintRunnerScript)&quot; -exclude &quot;$(TSLintExclude)&quot; -files-root-dir &quot;$(TSLintFilesRootDir)&quot; -file-list-file &quot;$(TSLintFileListFile)&quot;"
      IgnoreExitCode="true" />`

to 

`<Exec 
      Command="&quot;$(TSLintNodeExe)&quot; --harmony &quot;$(TSLintRunnerScript)&quot; -exclude &quot;$(TSLintExclude)&quot; -files-root-dir &quot;$(TSLintFilesRootDir)&quot; -file-list-file &quot;$(TSLintFileListFile)&quot;"
      IgnoreExitCode="false" />`

But when i do this get an error saying `node.exe exited with code -1`, and then i get the actual linting error. (Each as a row in VS error list.)

Do you know if it's possible to avoid the error about the exitcode and just show the linting error? Or is there a different approach to achieving this?

**Comment by JoshuaKGoldberg** [comment]
Hi!

I don't know the answer to that, likely due to having very little MSBuild experience.

Is having linting errors break build a feature you need?

**Comment by boedlen** [comment]
Having errors break the build, would force me and my colleagues to adhere to our rules all the time. If build doesn't break, it would be possible to have linting errors but still build and commit "broken" code.

So I'd say it's more of a `need` than `nice-to-have`. I will try to look into making ` IgnoreExitCode` a parameter and see if i can get rid of the `node.exe exited with code -1`, but i to have very limited experience with MSBuild.

**Comment by JoshuaKGoldberg** [comment]
#31 is a stopgap that lets the behavior you mentioned in.

**Comment by boedlen** [comment]
Hi Joshua,
Thanks for looking into this. I looked into this my self, and i think a found a different way to fix this problem, that also removes the "exited with code -1".

Here's my solutions that works on my machine
```xml
<?xml version="1.0" encoding="utf-8" ?>
<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <!-- Ensures that if this file changes it forces a TypeScript rebuild -->
  <PropertyGroup>
    <TypeScriptAllProjects>$(TypeScriptAllProjects);$(MSBuildThisFileFullPath)</TypeScriptAllProjects>
  </PropertyGroup>
  <Target 
    AfterTargets="CompileTypeScript"
    Condition="'@(TypeScriptCompile)' != ''"
    Name="TSLint">
    <PropertyGroup>
	  <TSLintBreakBuildOnError Condition="'$(TSLintBreakBuildOnError)' == ''>false</TSLintBreakBuildOnError>
      <TSLintDeleteFileListFile Condition="'$(TSLintDeleteFileListFile)' == ''">true</TSLintDeleteFileListFile>
      <TSLintExclude Condition="'$(TSLintExclude)' == ''">^$</TSLintExclude>
      <TSLintFilesRootDir Condition="'$(TSLintFilesRootDir)' == ''">$(MSBuildProjectDirectory)</TSLintFilesRootDir>
      <TSLintFileListDir Condition="'$(TSLintFileListDir)' == ''">$(IntermediateOutDir)</TSLintFileListDir>
      <TSLintFileListName Condition="'$(TSLintFileListName)' == ''">TSLintFileList-$(MSBuildProjectName).txt</TSLintFileListName>
      <TSLintFileListFile>$(TSLintFileListDir)$(TSLintFileListName)</TSLintFileListFile>
      <TSLintNodeExe Condition="'$(TSLintNodeExe)' == ''">$([System.IO.Path]::GetFullPath("$(MSBuildThisFileDirectory)\..\tools\node-5.9.0.exe"))</TSLintNodeExe>
      <TSLintRunnerScript Condition="'$(TSLintRunnerScript)' == ''">$([System.IO.Path]::GetFullPath("$(MSBuildThisFileDirectory)\..\tools\runner.js"))</TSLintRunnerScript>
    </PropertyGroup>

    <ItemGroup>
      <TSLintFile Include="@(TypeScriptCompile)" />
    </ItemGroup>

    <!-- Write file names to a text file so the runner can read them -->
    <MakeDir Directories="$(TSLintFileListDir)" />
    <WriteLinesToFile
      File="$(TSLintFileListFile)"
      Lines="@(TSLintFile->'%(Identity)')"
      Overwrite="true" />

    <!-- Run TSLint via the runner -->
    <Exec 
      Command="&quot;$(TSLintNodeExe)&quot; --harmony &quot;$(TSLintRunnerScript)&quot; -exclude &quot;$(TSLintExclude)&quot; -files-root-dir &quot;$(TSLintFilesRootDir)&quot; -file-list-file &quot;$(TSLintFileListFile)&quot;" IgnoreExitCode="true">
		<Output TaskParameter="ExitCode" PropertyName="ErrorCode"/>
	</Exec>
		
	<Message Importance="high" Text="'Exit code: $(ErrorCode)'"/>	
	
    <!-- Clean up our compiled file list when done -->
    <Delete Condition="'$(TSLintDeleteFileListFile)' == 'true'" Files="$(TSLintFileListFile)" />
	
    <!-- Return an Error if linter found errors and we should break on errors -->	
	<Error Condition="'$(ErrorCode)' == '-1' And '$(TSLintBreakBuildOnError)' == 'true'" />	
  </Target>
</Project>
```

**Comment by JoshuaKGoldberg** [comment]
That looks good, can you send a PR for review & testing?

**Comment by JoshuaKGoldberg** [comment]
Closed by #32

**Comment by JoshuaKGoldberg** [comment]
@boedlen How soon would you like this on nuget.org? I can have it up tonight, or if we wait until the weekend 2.0.1 will have a couple extra features (TBD what).
